### ISSUE TITLE
bundler on linux sometimes litters /tmp with empty dirs

### ISSUE THREAD

**Issue by antifuchs** [bundler/bundler#3366]
We recently ran out of disk space on `/tmp` in a machine, and found a huge number of directories of the form `bundler20150109-8649-jl0atf` and `bundler20150115-3605-akrvd7`. These are all empty, and we rmdir'd them (freeing up all the inodes), but it might point to a larger issue.

The versions of bundler we have observed this with are 1.6.2 and 1.7.12.

**Comment by indirect** [comment]
I vaguely recall noticing that sometimes there was a temporary directory that might not get deleted correctly. I’m not sure if I fixed it or not, though. Worth looking into.

**Comment by segiddins** [comment]
Seems to me that, on all four uses of `Bundler.tmp`, we don't do a good job of cleaning up after ourselves

**Comment by mikehale** [comment]
I've also been seeing this on 1.7.12.

**Comment by mikehale** [comment]
Actually I'm seeing files left over too. Stuff like this:

```console
/tmp/d20150504-18646-qokml8/bundler-1.7.12/bin
/tmp/d20150504-18646-qokml8/bundler-1.7.12/bin/bundler
/tmp/d20150504-18646-qokml8/bundler-1.7.12/bin/bundle
/tmp/d20150504-18646-qokml8/bundler-1.7.12/doc
/tmp/d20150504-18646-qokml8/bundler-1.7.12/cache
/tmp/d20150504-18646-qokml8/bundler-1.7.12/specifications
/tmp/d20150504-18646-qokml8/bundler-1.7.12/specifications/bundler-1.7.12.gemspec
```

**Comment by segiddins** [comment]
@mikehale those look like RubyGem's files leftover.

**Comment by segiddins** [comment]
This is probably fixed by https://github.com/bundler/bundler/pull/3277, which first shipped in 1.8.
